asllib/doc/ASLTypingReference.tex

```diff
@@ -8694,2385 +8629,4792 @@ \subsection{Comments}
-  \typearrow \vttwo
-}
-\and
-\inferrule[annotated]{}{
-  \annotateinittype(\tenv, \overname{\None}{\initialvaluetype}, \overname{\langle\vttwo\rangle}{\typeannotation})
-  \typearrow \vttwo
+\inferrule[single]{}{
+  \useslice(\overname{\SliceSingle(\ve)}{\vs}) \typearrow \overname{\useexpr(\ve)}{\ids}
 }
 \and
-\inferrule[initial]{}{
-  \annotateinittype(\tenv, \overname{\langle\vtone\rangle}{\initialvaluetype}, \overname{\None}{\typeannotation})
-  \typearrow \vtone
+\inferrule[star\_length\_range]{
+  L \in \{\SliceStar, \SliceLength, \SliceRange\}
+}{
+  \useslice(\overname{L(\veone, \vetwo)}{\vs}) \typearrow \overname{\useexpr(\veone) \cup \useexpr(\vetwo)}{\ids}
 }
 \end{mathpar}
 \end{emptyformal}
 
-\section{TypingRule.AddGlobalStorage \label{sec:TypingRule.AddGlobalStorage}}
-\hypertarget{def-addglobalstorage}{}
+\section{TypingRule.UseBitfield \label{sec:TypingRule.UseBitfield}}
+\hypertarget{def-usebitfield}{}
 The function
 \[
-  \addglobalstorage(
-    \overname{\staticenvs}{\tenv} \aslsep
-    \overname{\identifier}{\name} \aslsep
-    \overname{\GDkeyword}{\keyword} \aslsep
-    \overname{\ty}{\declaredt}
-  )
-  \aslto
-    \overname{\staticenvs}{\newtenv} \aslsep
-  \cup \overname{\TTypeError}{\TypeErrorConfig}
+\usebitfield(\overname{\decl}{\vbf}) \aslto \overname{\pow{\identifier}}{\ids}
 \]
-modifies the static environment $\tenv$ by adding a global storage
-for the identifier $\name$ with global storage keyword $\keyword$ and type $\declaredtypes$,
-resulting in the environment $\newtenv$.
-The result is a type error, if one is detected.
+returns set of identifiers $\ids$ on which the bitfield $\vbf$ depends on.
 
 \subsection{Prose}
-All of the following apply:
+One of the following applies:
 \begin{itemize}
-  \item checking that $\name$ is not declared in the global environment of $\tenv$ yields $\True$\ProseOrTypeError;
-  \item $\newtenv$ is $\tenv$ with its $\globalstoragetypes$ component updated by binding $\name$ to
-        $(\declaredt, \keyword)$.
+  \item All of the following apply (\textsc{simple}):
+  \begin{itemize}
+    \item $\vbf$ is the single field with slices $\slices$;
+    \item define $\ids$ as the union of applying $\useslice$ to each slice in $\slices$.
+  \end{itemize}
+
+  \item All of the following apply (\textsc{nested}):
+  \begin{itemize}
+    \item $\vbf$ is the nested bitfield with slices $\slices$ and bitfields $\bitfields$;
+    \item define $\ids$ as the union of applying $\useslice$ to each slice in $\slices$ and applying
+          $\usebitfield$ to each bitfield in $\bitfields$.
+  \end{itemize}
+
+  \item All of the following apply (\textsc{type}):
+  \begin{itemize}
+    \item $\vbf$ is the typed bitfield with slices $\slices$ and type $\tty$;
+    \item define $\ids$ as the union of applying $\useslice$ to each slice in $\slices$ and applying
+          $\usety$ to $\tty$.
+  \end{itemize}
 \end{itemize}
 
 \begin{emptyformal}
 
```

Comment:
catcher

Reply:
I don't understand the comment.

Reply:
sorry there is a typo: cathcer -> catcher